lib/kramdown-rfc2629.rb

```diff
@@ -42,32 +42,48 @@ def initialize(*doc)
         @block_parsers.unshift(:block_pi)
       end
 
-      SECTIONS_RE = /Section(?:s (?:[\w.]+, )*[\w.]+,? and)? [\w.]+/
-
-      def handle_bares(s, attr, format, href)
-        sa = s.sub(/\A\S+\s/, '').split(/,? and /)
-        sa[0..0] = *sa[0].split(', ')
-        sz = sa.size
-        if sz != 1         # we have to redo xml2rfc's work here
-          @tree.children << Element.new(:text, "Sections ", {}) # XXX needs to split into Section/Appendix
-          sa.each_with_index do |sec, i|
-            attr1 = {"target" => href, "section" => sec, "sectionFormat" => "bare"}
-            @tree.children << Element.new(:xref, nil, attr1)
-            text = if i == 0 && sz == 2
-                     " and "
-                   elsif i == sz-1
-                     " of "
-                   elsif i == sz-2
-                     ", and "
-                   else
-                     ", "
-                   end
-            @tree.children << Element.new(:text, text, {})
+      XREF_RE = /[!\?-]?[\w.-]+(?: \([^)]+\))?/
```

Comment:
`[-!?]` needs to be handled in a `xml_from_sections`, no?

`(?:([?!])(-)?|(-))` that is, actually.

Reply:
Yeah, that was an aborted attempt to work that out, after which I realized that the `[-!?]` mess doesn't recognize any of this stuff.  It's probably best to take that on separately.